python/cudf/cudf/core/index.py

```diff
@@ -1784,14 +1784,18 @@ def __init__(
             raise TypeError("Invalid dtype")
 
         if copy:
-            data = column.as_column(data).copy()
+            data = column.as_column(data, dtype=dtype).copy()
         kwargs = _setdefault_name(data, name=name)
         if isinstance(data, np.ndarray) and data.dtype.kind == "M":
             data = column.as_column(data)
         elif isinstance(data, pd.DatetimeIndex):
             data = column.as_column(data.values)
         elif isinstance(data, (list, tuple)):
             data = column.as_column(np.array(data, dtype=dtype))
+        elif isinstance(data, cudf.Series):
+            data = data._column.astype(dtype)
+        elif isinstance(data, column.ColumnBase):
+            data = data.astype(dtype)
```

Comment:
I feel like some of this logic that handles separate objects is already in `as_column`. How much of this still works if we just replace it with `data = as_column(data, dtype=dtype)`? Ideally we'd just have one funnel for this.

Reply:
Looks like `as_column` will handle everything that was already here. So I just deleted all the duplication here as re-directed everything to `as_column`.

Reply:
cc:@vyasr just FYI

Reply:
Yup I think that's fine. In the long run I anticipate that `as_column` (or a cleaned up version) will be used for constructors, but all other APIs will be much more prescriptive in what they accept. We can worry about `as_column` then, for now it's fine to increase our dependence on it.